src/inc/msquic.hpp

```diff
@@ -181,124 +181,124 @@ class UniquePtrArray {
     bool operator != (T* _ptr) const { return ptr != _ptr; }
 };
 
-class QuicApiTable : public QUIC_API_TABLE {
-    QUIC_STATUS Init;
-    const QUIC_API_TABLE* ApiTable{nullptr};
+class MsQuicApi : public QUIC_API_TABLE {
+    const QUIC_API_TABLE* ApiTable {nullptr};
+    QUIC_STATUS InitStatus;
 public:
-    QuicApiTable() noexcept {
-        if (QUIC_SUCCEEDED(Init = MsQuicOpen(&ApiTable))) {
+    MsQuicApi() noexcept {
+        if (QUIC_SUCCEEDED(InitStatus = MsQuicOpen(&ApiTable))) {
```

Comment:
Doesn't assignment within an if-statement cause C++ compilers to throw warnings?

Reply:
No, it doesn't. We do it in a ton of C++ places too.